app/plugin_api_test.go

```diff
@@ -510,6 +510,47 @@ func TestPluginAPILoadPluginConfigurationDefaults(t *testing.T) {
 	assert.Equal(t, "override35true", ret)
 }
 
+func TestPluginAPIGetBundlePath(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	setupPluginApiTest(t,
+		`
+		package main
+
+		import (
+			"github.com/mattermost/mattermost-server/plugin"
+			"github.com/mattermost/mattermost-server/model"
+		)
+
+		type MyPlugin struct {
+			plugin.MattermostPlugin
+		}
+
+		func (p *MyPlugin) MessageWillBePosted(c *plugin.Context, post *model.Post) (*model.Post, string) {
+			bundlePath, err := p.API.GetBundlePath()
+			if err != nil {
+				return nil, err.Error() + "failed get bundle path"
+			}
+
+			return nil, bundlePath
+		}
+
+		func main() {
+			plugin.ClientMain(&MyPlugin{})
+		}
+		`, `{"id": "testplugin", "backend": {"executable": "backend.exe"}}`, "testplugin", th.App)
+
+	hooks, err := th.App.GetPluginsEnvironment().HooksForPlugin("testplugin")
+	require.Nil(t, err)
+	require.NotNil(t, hooks)
+	bundlePath, err := filepath.Abs(filepath.Join(*th.App.Config().PluginSettings.Directory, "testplugin"))
```

Comment:
The test is a bit pointless, but I don't have a better idea.

Reply:
I was asking for it mostly so that we (a) maintain some level of commitment here to coverage; and (b) because this is the only compile-time compatibility test I think we have, for API signature via RPC. Thank you for doing this!

Reply:
I 100% agree with your mindset, but I'm a bit unhappy with the very limited way of properly testing the plugin API. On the other hand I don't have a good idea on how to make this better.

Reply:
Personally, I think this test is brilliant — it proves that the RPC API works. It’s exactly what I would have hoped for :)